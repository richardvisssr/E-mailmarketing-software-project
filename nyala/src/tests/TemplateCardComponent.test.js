import { cleanup, render } from "@testing-library/react";
import { createRoot } from "react-dom/client";
import { act } from "react-dom/test-utils";
import TemplateCard from "../components/adminpanel/templateCardComponent";
import { useRouter } from "next/navigation";

jest.mock("next/navigation", () => ({
  useRouter: jest.fn(),
}));

global.fetch = jest.fn(() =>
  Promise.resolve({
    json: () => Promise.resolve({ data: "mockData" }),
  })
);

afterEach(() => {
  fetch.mockClear();
  cleanup();
});

it("should render the template card component", async () => {
  // make the function async
  const div = document.createElement("div");
  const root = createRoot(div);

  await act(async () => {
    // use await act(async () => { ... })
    useRouter.mockImplementation(() => ({
      pathname: "",
      route: "",
      query: "",
      prefetch: jest.fn(),
    }));
    const mockTemplate = { id: "mockId" /* other properties as needed */ };
    root.render(<TemplateCard template={mockTemplate} />);
  });

  expect(root.container).toMatchSnapshot();
});
