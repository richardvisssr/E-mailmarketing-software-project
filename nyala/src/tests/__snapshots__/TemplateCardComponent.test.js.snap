// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the template card component 1`] = `undefined`;
